### DISCUSSION TITLE
Archiving Project

### DISCUSSION THREAD

**Discussion by tommoor** [outline/rich-markdown-editor#481] [General]
**After much hand-wringing I've decided to archive this project in the near future, it will no longer be maintained as part of the Outline organization.**

While the intentions of separately open sourcing the editor were noble, it has been problematic for a couple of key reasons – Outline as a product is built around the editor. Having a unique and tightly integrated editor experience is make or break for the Outline product, particularly as we enable collaborative editing. With this code living in a separate repository it adds a lot of overhead and decouples the UI and UX in a way that degrades the experience inside of Outline – for a single example, we're not able to use the same button or menu components.

I had originally hoped that open-sourcing the module might create a community around the editor that would help in getting bugs fixed and features launched, and while there have been a couple of contributions in the last year (for which I'm thankful!) for the most part it's been 99% me building this for free. Working with Prosemirror is also complicated enough that it doesn't make for good drive-by OSS contributions.

If you're looking for a flexible Prosemirror-based editor of the same vein I recommend checking out the following _excellent_ projects that surpass this one in many different ways:

- [TipTap](https://tiptap.dev/)
- [Remirror](https://remirror.io/)

There are also more than 300 forks of this project 😌

**Comment by marconunnari** [comment]
Hi Tom,

First of all thank you very much for developing this great editor. I like it because it's beautiful, comfortable to use and makes it really easy to use markdown (although TipTap is great, doesn't support markdown natively)

I totally understand your points but I'm a little sad that this project will be archived. I would happily sponsor you but I see that it's difficult to make money from Github sponsors. What about if you adopt a business model similar to others paid and open source editors like CKEditor and Froala?

**Reply by haveanicedavid** [reply]
Ah this is a bummer, I love the project but understand your dilemma.

Out of curiosity, are there any issues or contributions community members might provide which would result in changing your mind? I've gotten a lot of value out of the project and am slowly getting up to speed with the codebase (and Prosemirror, which is a bigger challenge) - would happily devote some of my time to giving back.

**Reply by tommoor** [reply]
It's hard to say... honestly just more fixes for some of the trickier outstanding issues and improvements, for example reordering table rows. 

If the editor for Outline does get moved inside the repo as I originally stated then this project will continue to exist, although may be less actively developed.

**Comment by maiya-22** [comment]
Oh, that's really disappointing; but you've got to take care of you.  I love this editor and would pay for it.  It would be great if there were a way to develop it for the public and get compensated.
